phoenix-core-client/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1415,8 +1425,23 @@ public List<Mutation> getMutationList() {
                             }, iwe, connection, connection.getQueryServices().getProps());
                             shouldRetryIndexedMutation = false;
                         } else {
-                            hTable.batch(mutationBatch, null);
+                            hTable.batch(mutationBatch, resultObjects);
+                        }
+
+                        if (connection.getAutoCommit()) {
+                            for (int i = 0; i < mutationBatch.size(); i++) {
+                                Result result = (Result) resultObjects[i];
+                                if (result != null && !result.isEmpty()) {
+                                    Cell cell = result.rawCells()[0];
+                                    numUpdatedRowsForAutoCommit = PInteger.INSTANCE.getCodec()
```

Comment:
Now we have 2 APIs on the MutationState object `getUpdateCount()` and `getNumUpdatedRowsForAutoCommit()`. They server a similar purpose but can give different results under some conditions. We should just keep the original API `getUpdateCount` which should handle all the conditions and give the correct result. 
One idea is we could introduce a new field called `numRowsIgnored` initialized to 0. Then if the Result cell is 0 that means the row is ignored and we increment `numRowsIgnored`.  Now in `getUpdateCount` we simply return `numRows - numRowsIgnored`

Reply:
`numRowsIgnored` is keeping track of the updates that are ignored

Reply:
Actually, we need to think more carefully about the client side changes. The code written so far is only handling auto-commit on a batch of size 1. It will not work when there is a batch of atomic upserts with auto-commit set to true. See [PHOENIX-6821](https://issues.apache.org/jira/browse/PHOENIX-6821)

Reply:
Discussed offline, we will only handle the case when auto-commit is set to true and the batch size is 1.